### PR TITLE
MDEV-19510 Issue with: sql/wsrep_mysqld.cc : ip_len

### PR DESCRIPTION
**Pull request by an3l**
This patch is done by Tony Reix, MCA.

### CODE REVIEW

File: sql/wsrep_mysqld.cc [hunk 1]
```diff
@@ -534,11 +534,11 @@ static std::string wsrep_server_incoming_address()
...
       {
-        size_t const ip_len= wsrep_host_len(node_addr.c_str(), node_addr.size());
-        if (ip_len + 7 /* :55555\0 */ < inc_addr_max)
+        size_t const ip_len_mdb= wsrep_host_len(node_addr.c_str(), node_addr.size());
+        if (ip_len_mdb + 7 /* :55555\0 */ < inc_addr_max)
         {
-          memcpy (inc_addr, node_addr.c_str(), ip_len);
-          snprintf(inc_addr + ip_len, inc_addr_max - ip_len, ":%u",
+          memcpy (inc_addr, node_addr.c_str(), ip_len_mdb);
+          snprintf(inc_addr + ip_len_mdb, inc_addr_max - ip_lenip_len_mdb":%u",
```
Comment (janlindstrom) [inline sql/wsrep_mysqld.cc]: Can you explain what was the issue with original ip_len and last line looks suspicious on ip_lenip_len_mdb ?

Comment (an3l) [reply sql/wsrep_mysqld.cc]: Please see jira MDEV description https://jira.mariadb.org/browse/MDEV-19510

**Comment by janlindstrom** [comment]
jan@jan-HP-ZBook-15u-G5:~/mysql/10.4$ gcc --version
gcc (Ubuntu 8.3.0-6ubuntu1) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

**Comment by cvicentiu** [comment]
@an3l The patch is ok, except you forgot the comma at the end when pushing it.

Please provide a bit more context in the commit message, as is visible in the mailing list. For example:

AIX netinet/ip.h header defines `ip_len` as `ip_ff.ip_flen`.  To prevent the define from overwriting code in MariaDB, rename the variable name to ip_len_mdb.

Additionally, find the earliest version that has this code and rebase the commit there. Then send it back to me for review one more time.

**Comment by an3l** [comment]
@cvicentiu patch updated!

**Comment by an3l** [comment]
Pushed with `ad5b7b157b4`.

**Pull request closed without merging**